addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200GatewayHandler.java

```diff
@@ -0,0 +1,712 @@
+            this.device = device;
+            this.comm = comm;
+        }
+
+        @Override
+        public void run() {
+            logger.debug("GetKM200Runnable");
+            synchronized (device) {
+                if (device.getInited()) {
+                    device.resetAllUpdates(device.serviceTreeMap);
+                    updateChildren(gatewayHandler, device, comm, null);
+                }
+            }
+        }
+    }
+
+    /**
+     * The GetKM200Runnable class get the data from device for one channel.
+     *
+     * @author Markus Eckhardt
+     *
+     * @since 2.1.0
+     */
+    private static class GetSingleKM200Runnable implements Runnable {
+
+        private final Logger logger = LoggerFactory.getLogger(GetSingleKM200Runnable.class);
+        private final KM200GatewayHandler gatewayHandler;
+        private final KM200Device device;
+        private final KM200Comm<KM200Device> comm;
+        private final Channel channel;
+
+        public GetSingleKM200Runnable(KM200GatewayHandler gatewayHandler, KM200Device device,
+                KM200Comm<KM200Device> comm, Channel channel) {
+            this.gatewayHandler = gatewayHandler;
+            this.device = device;
+            this.comm = comm;
+            this.channel = channel;
+        }
+
+        @Override
+        public void run() {
+            logger.debug("GetKM200Runnable");
+            State state = null;
+            synchronized (device) {
+                if (device.getInited()) {
+                    logger.debug("Checking: {} Root: {}", channel.getUID().getAsString(),
+                            channel.getProperties().get("root"));
+                    KM200CommObject object = device.getServiceObject(checkParameterReplacement(channel, device));
+                    if (object.getVirtual() == 1) {
+                        String parent = object.getParent();
+                        updateChildren(gatewayHandler, device, comm, parent);
+                    } else {
+                        object.setUpdated(false);
+                        state = comm.getProvidersState(checkParameterReplacement(channel, device),
+                                channel.getAcceptedItemType(), getChannelConfigurationStrings(channel));
+                        if (state != null) {
+                            try {
+                                gatewayHandler.updateState(channel.getUID(), state);
+                            } catch (IllegalStateException e) {
+                                logger.error("Could not get updated item state, Error: {}", e);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * The sendKM200Thread class sends the data to the device.
+     *
+     * @author Markus Eckhardt
+     *
+     * @since 2.1.0
+     */
+    private static class SendKM200Thread extends Thread {
+
+        private final Map<Channel, byte[]> sendMap;
```

Comment:
Every command has his own "place" in this array. If the user is changing it then this will be replaced here immediately. But this binding is sending only the last value with a sleep of some seconds to the device. It is a kind of spam protection.